google/cloud/pubsub/internal/tracing_message_batch_test.cc

```diff
@@ -204,11 +206,51 @@ TEST(TracingMessageBatch, FlushLargeBatch) {
   EXPECT_THAT(message_batch->GetBatchSinkSpans(), SizeIs(1));
 
   auto spans = span_catcher->GetSpans();
-  EXPECT_THAT(spans, Contains(AllOf(
-                         SpanHasInstrumentationScope(), SpanKindIsClient(),
-                         SpanNamed("BatchSink::AsyncPublish"),
-                         SpanHasAttributes(OTelAttribute<std::int64_t>(
-                             "messaging.pubsub.num_messages_in_batch", 129)))));
+  EXPECT_THAT(
+      spans, Contains(AllOf(SpanHasInstrumentationScope(), SpanKindIsClient(),
+                            SpanNamed("BatchSink::AsyncPublish"),
+                            SpanHasAttributes(OTelAttribute<std::int64_t>(
+                                "messaging.pubsub.num_messages_in_batch", 128)),
+                            SpanLinksSizeIs(128))));
+}
+
+TEST(TracingMessageBatch, FlushLargeBatch) {
+  int const k_batch_size = 129;
+  std::vector<opentelemetry::nostd::shared_ptr<opentelemetry::trace::Span>>
+      initial_spans;
+  initial_spans.reserve(k_batch_size);
+  for (int i = 0; i < k_batch_size; ++i) {
+    initial_spans.emplace_back(MakeSpan("test span " + std::to_string(i)));
+  }
```

Comment:
Consider refactoring this to a function... and maybe use `std::generate` or `std::generate_n`